breaking-changes.md

```diff
@@ -0,0 +1,13 @@
+
+# Breaking Changes
+
+## API
+
+- serve(options) -> serve(argv, options)
```

Comment:
1. It sounds strange when variable called `argv` expect an object, do you know any app or programming language where `argv` is not an array? If you still want an object, may be renamed the argument accordingly?
2. What the difference between first and second argument, they both accept object with almost the same options? Maybe just accept only one object will all possible configurations?
3. If the purpose of first argument is to accept command line arguments, maybe need to transform cli arguments to internal (object) format on webpack-serve side without the need for user to install something like minimist etc.?

Reply:
If you're going to use the API, you have to conform to what the API expects. The documentation is there to let you know what the right type is, and there's an error checking the type now as well. APIs doesn't take raw command line flags (unless the tool in question is explicitly for that purpose). I understand your concern but will not be changing that variable name at this time.

Reply:
Coming here from reading the 2.0.0 changelog - as someone who uses webpack-serve through the API, not the CLI, it seems like I have to always pass an empty object. I therefor don’t understand why this had to be introduced as required first parameter - seems like an unnecessary breaking change to me? Why not take it as a second optional parameter?

Reply:
I follow `options`  as the last argument, unless there's a callback, as a pattern. The four extra keystrokes (`{}, `) aren't that tough, come on now folks.

Reply:
I agree that `options` should always come after required parameters, but in this case `argv` seems like an _optional_ parameter, as the CLI is the only consumer needing to provide it and all other consumers of the API will pass an empty object - so I think it should have best been an option itself (a property of `options`).

No, it's not a big deal to type a few more characters, but it has an impact on the ecosystem, as we all cannot just merge automatic dependency update PRs (but have to manually checkout the branch, fix the API changes), we have to wait till the TypeScript typings are updated with the new signature, etc - causing slower adoption of the new version, which fragments the ecosystem. Basic consequences of _any_ type of breaking change.

But the ships have sailed anyway, so whatever - we'll just all update our code 😉